### DISCUSSION TITLE
JsonValue column type augmentation

### DISCUSSION THREAD

**Discussion by italodeandra** [prisma/prisma#7521] [Q&A]
Hi team,

I have this model:

```prisma
model User {
  info Json @default("{}")
}
```

Which generates this type:

```ts
export type User {
  info: Prisma.JsonValue
}
```

And when I use `await prisma.match.findMany` it will return an array of this type.

But let's say I want to make it return `info` as the actual type that I'm going to persist in there. How can I accomplish that with Prisma?

I tried the following:

```ts
declare module "@prisma/client" {
  interface User { // also tried type - and the `export` keyword is not required for augmentations
    info: {
      name?: string
    }
  }
}
```

But it didn't work. Probably because it is a `type` and not an `interface`? Does anyone have any tips on how can I accomplish this without needing to wrap all the usages of the generated methods with a utility that casts the type?

**Comment by ryands17** [comment]
@italodeandra 👋 

Yes this cannot be extended as it's a type. So you would need to explicitly cast it whenever you're using it. I would suggest following this request as it exactly matches your use case and will make typing Json values easier: https://github.com/prisma/prisma/issues/3219

**Reply by italodeandra** [reply]
Hey @ryands17, thanks for the reply.

Yeah, I figured it wasn't possible to augment a type. I do this a lot with interfaces though, so that's why I was wondering. I tried to look for something on Google, but this `augmentation` word is not well known yet.

#3219 seems more complicated, since the suggestion is to make possible to specify a type for the JSON from the schema itself. Do you think it makes sense to create another one suggesting for augmentation to be possible? For me that's exactly the purpose of augmentation, since it will be the dev's responsibility. We can't validate what a JsonValue column actually is from the database perspective, so it doesn't make sense to add that to the schema, in my opinion.

**Reply by ryands17** [reply]
I think mostly adding the type to the schema would be opt-in and if you know that you are setting a concrete type in the Json field then you could otherwise you would get the same old `Prisma.JsonValue`. By setting the types in TS, you are in turn assuming that the data coming from the Json field exactly matches the type.

You could add a comment in this request and a 👍 as once this would be implemented, then the types part will automatically be handled.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
